### DISCUSSION TITLE
How to do batch processing without running img2img

### DISCUSSION THREAD

**Discussion by ddPn08** [Mikubill/sd-webui-controlnet#153] [Ideas]
First, enable `Settings` > `ControlNet` > `Skip img2img processing when using img2img initial image` and `Do not append detectmap to output`.

![image](https://user-images.githubusercontent.com/71378929/219871652-faa5a666-8ca4-4846-b5df-069c674698fc.png)

Then open the `Batch` tab of the `img2img` tab.

Enter the folder containing the images for ControlNet in `Input directory`, and the path of the output destination folder in `Output directory`. Do not set images on the ControlNet canvas.

![image](https://user-images.githubusercontent.com/71378929/219870754-356b1e4b-e5a2-43d1-af3c-1491194e8e81.png)

**Comment by lyuro** [comment]
Queue, I want to know this too

**Comment by camenduru** [comment]
download from youtube and extract frames

```py
pip install yt_dlp
python -c "from yt_dlp import YoutubeDL; YoutubeDL({'overwrites':True, 'format':'bestvideo[ext=mp4]+bestaudio[ext=m4a]/mp4', 'outtmpl':'/content/video.mp4'}).download(['https://youtu.be/EU3hIXXeiz4'])"
mkdir /content/video
ffmpeg -i /content/video.mp4 '/content/video/%04d.png'
```
or

```py
pip install yt_dlp
python -c "import os; from yt_dlp import YoutubeDL; ydl_opts = {'overwrites':True, 'format':'bestvideo[ext=mp4]+bestaudio[ext=m4a]/mp4', 'outtmpl':'/content/video.mp4'}; YoutubeDL(ydl_opts).download(['https://youtu.be/EU3hIXXeiz4']); os.system('ffmpeg -i /content/video.mp4 \'/content/video/%04d.png\'')"
```
